File: 프로그래머스/lv0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        
                if (common[1] - common[0] == common[common.length - 1] - common[common.length - 2]) {
        	int diff = common[1] - common[0];
            return common[common.length - 1] + diff;
        } else if (common[1] / common[0] == common[common.length - 1] / common[common.length - 2]) {
        	int ratio = common[1] / common[0];
            return common[common.length - 1] * ratio;
        }
        
        return -1;
    }
}

File: 프로그래머스/unrated/120924. 다음에 올 숫자/다음에 올 숫자.js
function solution(common) {
    let answer = 0;
    const n = common.length;

    if (n < 2) {
        return -1;
    }

    const last = common[n - 1];
    const diff = common[1] - common[0];
    const ratio = common[1] / common[0];

    if (diff === common[n - 1] - common[n - 2]) {
        return last + diff;
    } else if (ratio === common[n - 1] / common[n - 2]) {
        return last * ratio;
    }

    return -1;
}
